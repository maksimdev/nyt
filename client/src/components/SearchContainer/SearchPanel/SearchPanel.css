.searchPanel {
  padding: 5%;
  width: 70%;
}

.searchPanel > .searchField {
  margin-bottom: 20px;  
}
