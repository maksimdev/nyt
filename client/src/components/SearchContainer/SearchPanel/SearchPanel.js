import React from 'react';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import './SearchPanel.css';

export function SearchPanel(props) {
    return (
      <Paper className="searchPanel">
        <TextField
            id="search"
            value={props.value}
            className="searchField"
            label="NYT"
            placeholder="Search articles"
            fullWidth
            variant="outlined"
            InputLabelProps={{
                shrink: true,
            }}
            onChange={props.textFieldHandler}
        />
        <Button variant="contained" color="primary" onClick={props.search}>
            Search
        </Button>
    </Paper>
    );
};