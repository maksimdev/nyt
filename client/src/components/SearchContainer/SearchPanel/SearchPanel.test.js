import React from 'react';
import * as Enzyme from 'enzyme';
import SearchPanel from './SearchPanel';
import Adapter from 'enzyme-adapter-react-16';
import enzymeSerializer from 'enzyme-to-json/serializer';

Enzyme.configure({ adapter: new Adapter() })

expect.addSnapshotSerializer(enzymeSerializer);
describe('SearchPanel', () => {
    it('should to match snapshot', () => {
        const SearchPanelComponent = Enzyme.shallow(<SearchPanel />);
        expect(SearchPanelComponent).toMatchSnapshot();
    });
});