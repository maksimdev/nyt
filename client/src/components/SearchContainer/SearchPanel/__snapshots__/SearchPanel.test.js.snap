// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel should to match snapshot 1`] = `
<WithStyles(Paper)
  className="searchPanel"
>
  <TextField
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    className="searchField"
    fullWidth={true}
    id="search"
    label="NYT"
    placeholder="Search articles"
    required={false}
    select={false}
    variant="outlined"
  />
  <WithStyles(Button)
    color="primary"
    variant="contained"
  >
    Search
  </WithStyles(Button)>
</WithStyles(Paper)>
`;
